pkg/process/processors/osv_processor.go

```diff
@@ -0,0 +1,73 @@
+package processors
+
+import (
+	"io"
+	"strings"
+
+	"github.com/anchore/grype-db/internal/log"
+	"github.com/anchore/grype-db/pkg/data"
+	"github.com/anchore/grype-db/pkg/provider"
+	"github.com/anchore/grype-db/pkg/provider/unmarshal"
+)
+
+type osvProcessor struct {
+	transformer any
+}
+
+func NewOSVProcessor(transformer data.OSVTransformer) data.Processor {
+	return &osvProcessor{
+		transformer: transformer,
+	}
+}
+
+func NewV2OSVProcessor(transformer data.OSTransformerV2) data.Processor {
+	return &osProcessor{
+		transformer: transformer,
+	}
+}
+
+func (p osvProcessor) Process(reader io.Reader, state provider.State) ([]data.Entry, error) {
+	var results []data.Entry
+
+	entries, err := unmarshal.OSVVulnerabilityEntries(reader)
+	if err != nil {
+		return nil, err
+	}
+
+	var handle func(entry unmarshal.OSVVulnerability) ([]data.Entry, error)
+	switch t := p.transformer.(type) {
+	case data.OSVTransformer:
+		handle = func(entry unmarshal.OSVVulnerability) ([]data.Entry, error) {
+			return t(entry)
+		}
+	case data.OSVTransformerV2:
+		handle = func(entry unmarshal.OSVVulnerability) ([]data.Entry, error) {
+			return t(entry, state)
+		}
+	}
+
+	for _, entry := range entries {
+		transformedEntries, err := handle(entry)
+		if err != nil {
+			return nil, err
+		}
+
+		results = append(results, transformedEntries...)
+	}
+
+	return results, nil
+}
+
+func (p osvProcessor) IsSupported(schemaURL string) bool {
+	matchesSchemaType := strings.Contains(schemaURL, "https://raw.githubusercontent.com/anchore/vunnel/main/schema/vulnerability/osv/schema-")
+	if !matchesSchemaType {
+		return false
+	}
+
+	if !strings.HasSuffix(schemaURL, "schema-1.6.1.json") {
```

Comment:
A lot (all?) of the actual data in bitnami vulndb is OSV schema version 1.5.0. Shouldn't this accept 1.5.0 schema versions?

Reply:
It has to, because when merging anchore/vunnel#512, we changed the code so that instead of claiming every record was a 1.6.1 OSV schema, we put the actual schema from the record in each record.

This means that grype-db can handle 1.5.0 and 1.6.1 records _differently_ if it needs to, but it also means that it certainly needs to accept 1.5.0.

Reply:
@juan131 I'll just push a change really fast so that we can get this merged.

Reply:
@juan131 what do you think about supporting any `1.x.x` schema? That's what @wagoodman and I decided to allow the vunnel provider to emit in anchore/vunnel#512, so I think that's the way to go here.

Also, I didn't realize you were actively working on this right now - I'll hold off working on this until later today so I don't make extra work for you with merge conflicts.

Reply:
I agree on supporting `1.x.x` schema @willmurphyscode 

By the way, feel free to resume the work, I'm leaving in an hour and then I'll be on PTO until Monday. Thanks for your collaboration by the way.